### PR TITLE
feat: ticking gapminder dataset

### PR DESCRIPTION
**Pull request by alexpeters1208**
Still need to do the docstring and consider whether perf can be improved

### CODE REVIEW

File: plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py [hunk 1]
```diff
@@ -404,3 +405,190 @@ def generate_tip(total_bill: float, index: int) -> float:
...
+    )
+
+    ### First, we're going to construct the expanded, interpolated dataset
+
+    # functions for producing lists to expand original dataset
+    def create_years(year: int, reps: int) -> typing.List[int]:
+        return [year + i for i in range(reps)]
+
+    def create_empty(val: typing.Any, reps: int) -> typing.List[typing.Any]:
+        return [val, *[np.nan for _ in range(reps - 1)]]
```
Comment (jnumainville) [inline plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:437]: since we use `from __future__ import annotations` you can use `list[int]` instead of `typing.List[int]`
also, prefer just `Any` to `typing.Any`

File: plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py [hunk 2]
```diff
@@ -404,3 +405,190 @@ def generate_tip(total_bill: float, index: int) -> float:
...
+
+    # split gapminder into pre-2007 and 2007, since 2007 need not be expanded in the same way as previous years
+    gapminder_no_2007 = typing.cast(pd.DataFrame, gapminder[gapminder["year"] != 2007])
+    gapminder_2007 = typing.cast(pd.DataFrame, gapminder[gapminder["year"] == 2007])
```
Comment (jnumainville) [inline plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:441]: It it possible to check if `isinstance(pd.DataFrame, ...)` instead of cast? Better to use `typing.cast` sparingly
this is also a general comment for all `typing.cast` calls in this generator

File: plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py [hunk 3]
```diff
@@ -404,3 +405,190 @@ def generate_tip(total_bill: float, index: int) -> float:
...
+        column=["year", "lifeExp", "pop", "gdpPercap"]
+    )
+
+    # insert month column, where each element is a list of all 12 months
+    gapminder_no_2007.insert(
+        loc=3,
+        column="month",
+        value=typing.cast(
+            pd.Series,
+            [[month for month in range(1, 13)] for _ in range(len(gapminder_no_2007))],
```
Comment (jnumainville) [inline plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:460]: Recommend moving this to another function - this complex of a list comprehension can be hard to parse

File: plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py [hunk 4]
```diff
@@ -404,3 +405,190 @@ def generate_tip(total_bill: float, index: int) -> float:
...
+        (
+            gapminder_combined.groupby("country", observed=True, as_index=True)[
+                ["lifeExp", "pop", "gdpPercap"]
+            ]
+            .apply(lambda country: country.interpolate(method="linear"))
+            .reset_index(drop=True)
+        ),
+    )
+    gapminder_interp_vals.loc[:, "pop"] = gapminder_interp_vals["pop"].apply(
+        lambda x: round(x)
```
Comment (jnumainville) [inline plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:509]: `lambda x: round(x)` is equivalent to just passing round in this case
```suggestion
        round
```

**Review by jnumainville — CHANGES_REQUESTED** [review]

File: plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py (hunk 1 above)
Comment (alexpeters1208) [reply plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:437]: The `Any` change gives me this:
```
/Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:436:27 - error: "Any" is not defined (reportUndefinedVariable)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:436:51 - error: "Any" is not defined (reportUndefinedVariable)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:566:20 - error: "uby" is not defined (reportUndefinedVariable)
/Users/alexpeters/Documents/Web/deephaven-plugins/plugins/ui/src/deephaven/ui/_internal/RenderContext.py

```

File: plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py (hunk 2 above)
Comment (alexpeters1208) [reply plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:441]: The typing cast is the only way I managed to get around the typechecker. The result is definitely a pd.DataFrame by construction (and I checked it manually in several places), but Pyright and Pandas seem to not always get along. For example, this particular change gives:
```
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:445:68 - error: Cannot access member "apply" for type "ndarray[Any, Unknown]"
    Member "apply" is unknown (reportGeneralTypeIssues)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:445:68 - error: Cannot access member "apply" for type "NDArray[Unknown]"
    Member "apply" is unknown (reportGeneralTypeIssues)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:445:68 - error: Cannot access member "apply" for type "NDArray[Any]"
    Member "apply" is unknown (reportGeneralTypeIssues)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:445:23 - error: Cannot access member "loc" for type "ndarray[Any, Unknown]"
    Member "loc" is unknown (reportGeneralTypeIssues)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:445:23 - error: Cannot access member "loc" for type "NDArray[Unknown]"
    Member "loc" is unknown (reportGeneralTypeIssues)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:450:7 - error: Cannot access member "map" for type "ndarray[Any, Unknown]"
    Member "map" is unknown (reportGeneralTypeIssues)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:450:7 - error: Cannot access member "map" for type "NDArray[Unknown]"
    Member "map" is unknown (reportGeneralTypeIssues)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:450:7 - error: Cannot access member "map" for type "ndarray[Any, _dtype]"
    Member "map" is unknown (reportGeneralTypeIssues)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:448:23 - error: Cannot access member "loc" for type "ndarray[Any, Unknown]"
    Member "loc" is unknown (reportGeneralTypeIssues)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:448:23 - error: Cannot access member "loc" for type "NDArray[Unknown]"
    Member "loc" is unknown (reportGeneralTypeIssues)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:451:43 - error: Cannot access member "explode" for type "ndarray[Any, Unknown]"
    Member "explode" is unknown (reportGeneralTypeIssues)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:451:43 - error: Cannot access member "explode" for type "NDArray[Unknown]"
    Member "explode" is unknown (reportGeneralTypeIssues)
  /Users/alexpeters/Documents/Web/deephaven-plugins/plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:452:9 - error: No parameter named "column" (reportGeneralTypeIssues)
```
The type checker is thinking it's a numpy array. I guess dataframes are backed by numpy arrays in many cases, but the type of `gapminder_no_2007` is certainly a dataframe.

File: plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py (hunk 1 above)
Comment (jnumainville) [reply plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py]: did you import `Any` directly as well? don't see the import but not sure if that's just because you didn't commit the new import because that error came up
`from typing import Any`

File: plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py (hunk 2 above)
Comment (jnumainville) [reply plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py:445]: very odd - not sure what's going on in particular here, maybe a pandas issue

File: plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py (hunk 1 above)
Comment (alexpeters1208) [reply plugins/plotly-express/src/deephaven/plot/express/data/data_generators.py]: Did not catch this, I would have thought that the `from __future__ import annotations` would have had this. I don't know a ton about Python type hinting.

**Review by jnumainville — DISMISSED** [review]

**Review by jnumainville — APPROVED** [review]
